helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -242,12 +242,32 @@ receivers:
           - move:
               from: log
               to: $$
-
   {{- if .Values.logsCollection.extraFileLogs }}
   {{- toYaml .Values.logsCollection.extraFileLogs | nindent 2 }}
   {{- end }}
+  # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver
+  {{- if (eq (include "splunk-otel-collector.logsEnabled" .) "true") }}
+  {{- if and (eq .Values.logsEngine "otel") .Values.logsCollection.journald.enabled }}
+  {{- if .Values.logsCollection.journald.units }}
+  {{- range $_, $unit := .Values.logsCollection.journald.units }}
+  {{- printf "journald/%s:" $unit.name | nindent 2 }}
+    directory: {{ $.Values.logsCollection.journald.directory }}
+    units: [{{ $unit.name }}]
+    priority: {{ $unit.priority | default $.Values.logsCollection.journald.defaultPriority }}
+    resource:
+      com.splunk.source: {{ $.Values.logsCollection.journald.directory }}
+      com.splunk.sourcetype: 'EXPR("kube:"+$$._SYSTEMD_UNIT)'
+      com.splunk.index: {{ $.Values.logsCollection.journald.index | default $.Values.splunkPlatform.index}}
+      host.name: 'EXPR(env("K8S_NODE_NAME"))'
+  {{- end }}
+  {{- else }}
+  journald:
+    directory: {{- toYaml .Values.logsCollection.journald.directory | nindent 6 }}
+    priority: {{ .Values.logsCollection.journald.defaultPriority }}
```

Comment:
How to get into this condition? By setting `logsCollection.journald.units` to `null`? Will this collect logs from all available systemd services in that case?

May be this can be default and we provide a way to configure `units` as an option?

Please document it

Reply:
Nvm. Just found out all the answers, and it is documented